api/router.js

```diff
@@ -6,31 +6,45 @@ var autoredraw = require("../api/autoredraw")
 
 module.exports = function($window, renderer, pubsub) {
 	var router = coreRouter($window)
+	var identity = function(o){return o}
+	var globalId
+	var routeRequested = false
+	function onbeforeresolve(){routeRequested = true}
+
 	var route = function(root, defaultRoute, routes) {
-		var current = {path: null, component: "div"}
+		var current = {path: null, component: "div", render:identity}
 		var replay = router.defineRoutes(routes, function(payload, args, path, route) {
-			args.path = path, args.route = route
-			if (typeof payload.onmatch === "function") {
-				if (typeof payload.view !== "function") payload.view = function(vnode) {return vnode}
```

Comment:
If this condition resolves, we're looking at

```js
{
  onmatch : Function,
  view : x => x
}
```

I'm still not entirely sure how the various RouteResolver behaviours are meant to work together, but… This is wrong, right?